cocos2d/core/base-nodes/CCNodeCanvasRenderCmd.js

```diff
@@ -368,24 +371,50 @@ cc.Node.RenderCmd.prototype = {
     },
 
     visitChildren: function(){
```

Comment:
With proper z values for all the nodes, it would be possible to render some nodes out-of-order and therefor do more batching. For example, if you have sets of sprites with opaque textures that can't be batched because they are mixed in the render order, you could still batch them and it would end up looking correct because the z buffer values make sure that they end up in the correct order in the depth buffer. However, if you want to keep node draw order 100% correct (i.e. renderCmds are always drawn in the order in which they appear in the renderCmdList of the renderer) then this is not necessary.